hypershift-operator/controllers/hci/aws_infra.go

```diff
@@ -0,0 +1,169 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hostedclusterinfrastructure
+
+import (
+	awssdk "github.com/aws/aws-sdk-go/aws"
+	"github.com/openshift/hypershift/api/util/ipnet"
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/cmd/infra/aws"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Returns true when a requeue is required and pass an error if it is available
+// We only reconcile Infra or IAM if it isn't already AsExpected
+// The Status patches are organized to make sure we don't lose status data
+func (r *HostedClusterInfrastructureReconciler) createAws(hci *hyperv1.HostedClusterInfrastructure, ps *corev1.Secret) (bool, error) {
+	oHci := hci.DeepCopy()
+	_, infraStatus, iamStatus := GetInfraIamConditionStatus(hci)
+	if !infraStatus {
+
+		if err := r.updateStatusConditionsOnChange(hci, oHci, hyperv1.PlatformConfigured, metav1.ConditionFalse, "Configuring Infra with infra-id: "+hci.Spec.InfraID, hyperv1.HCIBeingConfiguredReason); err != nil {
+			return true, err
+		}
+		if rq, err := r.createAwsInfra(hci, ps); rq {
```

Comment:
what is rq? if it's requeue I think that is something which this function should not be aware of or reference here, It should be up to the upper level consumer function to decide what to do with the output of this one, whether that's requeing, erroring of whatever is up to that consumer.

Reply:
Originally I thought I had remembered a few cases where we hit an error and I didn't want to requeue, so that would be false, but with an error. Cases where the error requires you to change the HCI for example, vs something that would need to be fixed in AWS where a requeue would be needed to detect the change. Scanning the functions, I don't see  me ever returning false with an error, so I'll move it up to the caller as you suggested.

Reply:
Sorry as I was deleting these and I remembered now, most of these failures we want to requeue, at 30s and not immediately. The only time we requeue with an error is if the update failed.

I'll make the change as so, if there is an error it will requeue in 30s.  I'll just log the error from the update condition.

Reply:
no longer applies to this pr.